packages/cosmo-dycore/package.py

```diff
@@ -74,16 +75,11 @@ class CosmoDycore(CMakePackage):
 
     root_cmakelists_dir='dycore'
 
-    def setup_environment(self, spack_env, run_env):
-        if self.spec['mpi'].name == 'mpich':
-            spack_env.set('MPICH_G2G_PIPELINE', '64')
+    def setup_run_environment(self, env):
+        if '~cuda' in self.spec and self.spec['mpi'].name == 'mpich':
+            env.set('MPICH_G2G_PIPELINE', '64')
             if '+cuda' in self.spec:
```

Comment:
sure about indentation of the if, I think you will never go in the +cuda it is included in a ~cuda no, also I think the first one "MPICH_G2G_PIPELINE" (this refers to a GPU 2 GPU configuration) should be in the +cuda

Reply:
Yeah and I should actually put that in the mpich package, not dycore, thx!